pkg/build/admission/defaults/admission.go

```diff
@@ -71,6 +71,28 @@ func (b BuildDefaults) applyPodDefaults(pod *kapi.Pod) {
 	for k, v := range b.config.Annotations {
 		addDefaultAnnotations(k, v, pod.Annotations)
 	}
+
+	// Apply default resources
+	defaultResources := b.config.Resources
+	if len(pod.Spec.Containers[0].Resources.Limits) == 0 {
```

Comment:
Looks like the auto conversion logic handles it.  From https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/api/v1/zz_generated.conversion.go#L3213

`out.Containers = *(*[]Container)(unsafe.Pointer(&in.Containers))`